src/accel/LocalOffloadInterface.cc

```diff
@@ -0,0 +1,114 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file accel/LocalOffloadInterface.cc
+//---------------------------------------------------------------------------//
+#include "LocalOffloadInterface.hh"
+
+#include <G4EventManager.hh>
+#include <G4MTRunManager.hh>
+#include <G4Threading.hh>
+
+#ifdef _OPENMP
+#    include <omp.h>
+#endif
+
+#include "corecel/io/Logger.hh"
+#include "corecel/sys/Device.hh"
+#include "corecel/sys/Environment.hh"
+#include "geocel/GeantUtils.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Set the event ID and reseed the Celeritas RNG at the start of an event.
+ */
+void LocalOffloadBase::InitializeEvent(int id)
+{
+    CELER_EXPECT(id >= 0);
+
+    event_id_ = id_cast<UniqueEventId>(id);
+
+    if (!(G4Threading::IsMultithreadedApplication()
+          && G4MTRunManager::SeedOncePerCommunication()))
+    {
+        // Since Geant4 schedules events dynamically, reseed the Celeritas RNGs
+        // using the Geant4 event ID for reproducibility. This guarantees that
+        // an event can be reproduced given the event ID.
+        this->Reseed(event_id_.get());
```

Comment:
I forgot this was here when I commented previously: what if you make the interface for this function and below:
```c++
// Return new event ID, or null ID if the same event is already running
[[nodiscard]]  UniqueEventId update(G4Event*);
// Update from run manager, returning a new event ID if updated
[[nodiscard]]  UniqueEventId update();
```

and then the caller can do:
```c++
if (auto new_id = local_event_.update(g4event))
{
    state_.reseed(...);
}
```
for the explicit "initialize event" or 

```c++
if (auto new_id = local_event_.update())
{
    state_.reseed(...);
}
```
for the "flush".

Reply:
We'll still need to implement `LocalOffloadInterface::InitializeEvent(int)`... maybe I'll just duplicate this code for now and we can come back to it.